lucene/core/src/java/org/apache/lucene/util/hnsw/NeighborArray.java

```diff
@@ -111,6 +129,12 @@ public int[] sort() {
   private int insertSortedInternal() {
     assert sortedNodeSize < size : "Call this method only when there's unsorted node";
     int tmpNode = node[sortedNodeSize];
+    // Check if we need to compute score
+    if (scoringContext.containsKey(sortedNodeSize)) {
+      score[sortedNodeSize] = scoringContext.get(sortedNodeSize).calculateScore();
+      scoringContext.remove(sortedNodeSize);
+    }
```

Comment:
You should simplify and only call the map once.
```suggestion
    // Check if we need to compute score
    ScoringFunction maybeScoringFunction = scoringContext.remove(sortedNodeSize);
    if (maybeScoringFunction != null) {
      score[sortedNodeSize] = maybeScoringFunction.calculateScore();
    }
```